pkg/build/generator/generator.go

```diff
@@ -416,6 +416,20 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	build.Labels[buildapi.BuildConfigLabelDeprecated] = bcCopy.Name
 	build.Labels[buildapi.BuildConfigLabel] = bcCopy.Name
 
+	// Set the queued label for builds created from "serial" and "rapid" build configuration.
+	// The "serial" scheduling policy is default policy if not specified.
+	policy := bc.Spec.SchedulePolicy
+	if policy == nil {
+		build.Labels[buildapi.BuildQueuedLabel] = "true"
+	} else {
+		switch *policy {
+		case buildapi.BuildSchedulePolicySerial:
+			build.Labels[buildapi.BuildQueuedLabel] = "true"
+		case buildapi.BuildSchedulePolicySerialLatestOnly:
+			build.Labels[buildapi.BuildQueuedLabel] = "false"
```

Comment:
@bparees the "false" is confusing here but I don't want to introduce another label for "latest-only" builds... What I tried to say here is that when the build is latest-only, it will not be queued, but executed next (queued=false). The middle one will never run if you create third one, but until you create the third one, it is the one that will run next...